module-db/tests/ContactsRecord_tests.cpp

```diff
@@ -760,3 +760,163 @@ TEST_CASE("Check if new contact record exists in DB as a temporary contact")
+
+    std::array<std::string, 2> numbers   = {{{"500600100"}, {"500600200"}}};
+    std::array<std::string, 2> numbersPL = {{{"+48500600100"}, {"+48500600200"}}}; // Poland country code
+    std::array<std::string, 2> numbersFR = {{{"+33500600100"}, {"+33500600200"}}}; // France country code
+
+    SECTION("Temporary contact exists but new one with same number have country code")
+    {
+        // Number of record in number table should stay the same
+        constexpr int32_t NUMBER_IN_RECORD_DURING_REPLACEMENT = 1;
+
+        // Adding of temporary contact without country code
+        ContactRecord temporaryContactRecord;
+        temporaryContactRecord.numbers = std::vector<ContactRecord::Number>({
+            ContactRecord::Number(numbers[0], std::string(""), ContactNumberType::HOME),
+        });
+        temporaryContactRecord.addToGroup(contactsDb.get().groups.temporaryId());
+
+        REQUIRE(records.Add(temporaryContactRecord));
+        REQUIRE(contactsDb.get().number.count() == NUMBER_IN_RECORD_DURING_REPLACEMENT);
+
+        auto validationRecord = records.GetByID(1); // There aren't new normal contact
+        REQUIRE(validationRecord.ID == DB_ID_NONE);
+        REQUIRE(validationRecord.numbers.empty());
+
+        validationRecord = records.GetByIdWithTemporary(1); // There are new temporary contact
+        REQUIRE(validationRecord.numbers.size() == 1);
+        REQUIRE(validationRecord.numbers[0].number.getEntered() == numbers[0]);
+
+        // Adding of normal contact witch same number than previous temporary but with country code
+        ContactRecord noTemporaryContactRecord;
+        noTemporaryContactRecord.primaryName     = "PrimaryNameNoTemporary";
+        noTemporaryContactRecord.alternativeName = "AlternativeNameNoTemporary";
+        noTemporaryContactRecord.numbers         = std::vector<ContactRecord::Number>({
+            ContactRecord::Number(numbersPL[0], std::string(""), ContactNumberType::HOME),
+        });
+
+        REQUIRE(records.verifyTemporary(noTemporaryContactRecord) == true);
+
+        REQUIRE(records.Add(noTemporaryContactRecord));
+        REQUIRE(contactsDb.get().number.count() == NUMBER_IN_RECORD_DURING_REPLACEMENT); // That was replacement in
+                                                                                         // place
+
+        validationRecord = records.GetByID(1);
+        REQUIRE(validationRecord.numbers.size() == 0); // Now first contact is temporary
+        REQUIRE(validationRecord.ID == DB_ID_NONE);
+        REQUIRE(validationRecord.numbers.empty());
+
+        validationRecord = records.GetByID(2);
+        REQUIRE(validationRecord.numbers.size() == 1);
+        REQUIRE(validationRecord.numbers[0].number.getEntered() == numbersPL[0]);
+    }
+
+    SECTION("Adding contact without country code after deleting one with same number but with country code")
+    {
+        // Scenario from MOS-864
+        // Number of record in number table should stay the same
```

Comment:
Because not everyone has access to the MOS-864 JIRA ticket it would be good to shortly describe this scenario in this place.

Reply:
You right. I added some description there. Let me know if that's clear to you :D

Reply:
I had something else on my mind:
1. Create contact with the number +XX YYY YYY YYY
2. Delete that contact
3. Create a new contact with the the same number YYY YYY YYY but without country prefix code
4. Check if the new contact has the number provided in step 3. 
That short description is ok, but it would be perfect to have a generic scenario written as steps to execute :)

Reply:
Ok, I'll keep it mind. Thank you.
I leave this as it is for now.

Is there any other things to correct?

Reply:
No, it looks good now :)